app/policies/admin/generate_reference_policy.rb

```diff
@@ -0,0 +1,31 @@
+module Admin
+  class GenerateReferencePolicy < ApplicationPolicy
+    class Scope < Struct.new(:user, :scope)
+      def resolve
+        scope
+      end
+    end
+
+    def index?
+      user.is_admin? || user.permission_names.include?('create_generate_references')
+    end
+
+    def create?
+      user.is_admin? || user.permission_names.include?('create_generate_references')
```

Comment:
We will need a migration API side to create these new permissions so please do not merge until this is done in the API side

Reply:
I was more efficient than I thought - I had already done it before you started the admin.  Trouble is, I forgot !!

So, the permission names are different - please can you change to 

read_reference_number_generators and create_reference_number_generators - I guess you will only need to use the latter ?